frontend/javascripts/oxalis/controller/url_manager.js

```diff
@@ -181,6 +195,20 @@ class UrlManager {
         stateByLayer[layerName] = { mappingInfo: { mappingName, mappingType } };
       }
     }
+    for (const layerName of Object.keys(state.localSegmentationData)) {
+      const { isosurfaces, currentMeshFile } = state.localSegmentationData[layerName];
+      if (currentMeshFile != null) {
+        const { meshFileName } = currentMeshFile;
+        const meshes = Utils.values(isosurfaces)
+          .filter(({ isVisible }: IsosurfaceInformation) => isVisible)
+          .map(({ segmentId, seedPosition, isPrecomputed }: IsosurfaceInformation) => ({
```

Comment:
I added these types because of this observation:

> A couple of times I specified the type if the values were mapped, afterwards, which is a separate issue, since mapping `Array<number>` will lead to flow assuming a type of `number | any` for the mapped value argument :shrug: I did not systematically look for these occurrences, but I assume there are quite a few in our code, leading to less type security.

However, I revisited the type annotations and found out that the `any`s were not caused by the map, but instead because `stateByLayer` was initialized with `{}` without a type annotation. This led to flow inferring less-restrictive types. Adding the correct type annotation `UrlStateByLayer` to the initialization fixes this. So, as you say, these types can be removed and mapping doesn't produce `any`s :+1:

Reply:
Ah, makes sense. Thanks for the explanation!